src/test/versionbits_tests.cpp

```diff
@@ -443,8 +417,18 @@ BOOST_AUTO_TEST_CASE(versionbits_computeblockversion)
     // ACTIVE and FAILED states in roughly the way we expect
     for (const auto& chain_name : {CBaseChainParams::MAIN, CBaseChainParams::TESTNET, CBaseChainParams::SIGNET, CBaseChainParams::REGTEST}) {
         const auto chainParams = CreateChainParams(*m_node.args, chain_name);
+        std::set<decltype(Consensus::BIP9Deployment::bit)> chain_bits;
+        std::set<uint32_t> chain_masks;
         for (int i = 0; i < (int)Consensus::MAX_VERSION_BITS_DEPLOYMENTS; ++i) {
-            check_computeblockversion(chainParams->GetConsensus(), static_cast<Consensus::DeploymentPos>(i));
+            const auto dep = static_cast<Consensus::DeploymentPos>(i);
+            // Check that no bits are re-used (within the same chain). This is
+            // disallowed because the transition to FAILED (on timeout) does
+            // not take precedence over STARTED/LOCKED_IN. So all softforks on
+            // the same bit might overlap, even when non-overlapping start-end
+            // times are picked.
+            BOOST_CHECK(chain_bits.insert(chainParams->GetConsensus().vDeployments[dep].bit).second);
+            BOOST_CHECK(chain_masks.insert(VersionBitsMask(chainParams->GetConsensus(), dep)).second);
```

Comment:
Rather than adding to a `std::set<uint32_t>`, should we check set `chain_masks` to a uint32_t, and then check that there's no overlap between the signaled bits:

```diff
--- a/src/test/versionbits_tests.cpp
+++ b/src/test/versionbits_tests.cpp
@@ -418,7 +418,7 @@ BOOST_AUTO_TEST_CASE(versionbits_computeblockversion)
     for (const auto& chain_name : {CBaseChainParams::MAIN, CBaseChainParams::TESTNET, CBaseChainParams::SIGNET, CBaseChainParams::REGTEST}) {
         const auto chainParams = CreateChainParams(*m_node.args, chain_name);
         std::set<decltype(Consensus::BIP9Deployment::bit)> chain_bits;
-        std::set<uint32_t> chain_masks;
+        uint32_t chain_masks{0};
         for (int i = 0; i < (int)Consensus::MAX_VERSION_BITS_DEPLOYMENTS; ++i) {
             const auto dep = static_cast<Consensus::DeploymentPos>(i);
             // Check that no bits are re-used (within the same chain). This is
@@ -427,7 +427,9 @@ BOOST_AUTO_TEST_CASE(versionbits_computeblockversion)
             // the same bit might overlap, even when non-overlapping start-end
             // times are picked.
             BOOST_CHECK(chain_bits.insert(chainParams->GetConsensus().vDeployments[dep].bit).second);
-            BOOST_CHECK(chain_masks.insert(VersionBitsMask(chainParams->GetConsensus(), dep)).second);
+            const uint32_t chain_mask = VersionBitsMask(chainParams->GetConsensus(), dep);
+            BOOST_CHECK(!(chain_masks & chain_mask));
+            chain_masks |= chain_mask;
             check_computeblockversion(chainParams->GetConsensus(), dep);
         }
     }
```

Reply:
Thanks, done that.